src/StatusBar.ts

```diff
@@ -41,32 +77,156 @@ export class StatusBar {
       },
     }
   }
+
+  onDidChangeActiveTextEditor(editor: vscode.TextEditor) {
+    if (editor && editor.document) {
+      const folder = vscode.workspace.getWorkspaceFolder(editor.document.uri)
+      if (folder && folder.name !== this.activeFolder) {
+        this._activeFolder = folder.name
+        this.updateActiveStatusItem()
+      }
+    }
+  }
+
   private enqueue(source: string, status: Status, details?: string) {
     this.queue = this.queue.filter(x => x.source !== source)
-    this.queue.unshift({
+    const item: QueueItem = {
       source,
       status,
       details,
-    })
-    this.updateStatus()
+    }
+    this.queue.unshift(item)

```

Comment:
i just realize, that there is a memory leakage here. We should filter out all previous items with same source. Would you mind to add it?

Reply:
I think we can be more efficient with a Map, not array/queue here. Since we really just need 1 status per source and the sequence is not important since you used the status priority as the determining factor... take a look of the updated StatusBar and see if it makes sense...

Reply:
@escaton did you get the chance to look at the updated file? does that address your request?